### PR TITLE
Refactor the implementation for getting status of rebalance proposal request

### LINKED ISSUE #10294: Getting user task status of getting proposal and current rebalancing should be "consistent"
When using the `KafkaRebalance` custom resource for rebalancing purposes, we are using two different approaches for getting the status of:

* getting a proposal (CC is computing an optimization proposal)
* actual rebalancing (CC is moving data across the cluster)

The main CC REST endpoint is `/rebalance` which is used for both, getting a proposal (by using the query parameter `dry_run=true`, with no user task id) and starting the rebalancing (by using the query parameter `dry_run=false`, with no user task id).
In both cases, an HTTP POST request is issued to that endpoint which is right.

When it comes to get the status, we are:

* still issuing an HTTP POST request on `/rebalance?dry_run=true` for the status of getting a proposal but this time passing a user task id.
* issuing an HTTP GET request on `/user_tasks` for the status of the rebalancing (of course, including the user task id).

The only reason I can think of why it was done differently is because of the following specific case.
When the model on CC is not ready yet, so you have the usual exception about "not enough data in the window" (i.e. when CC has been started from a few minutes and it doesn't have enough data to process proposal), what happens on CC is that your POST request for a proposal is created on CC, it gets a user task id assigned, the exception is thrown, and the task is immediately removed from the tasks list. It means, that after the POST request, there is no status to get because that user task doesn't exist anymore so we are safe by issuing a new POST request which actually means requesting for a new proposal (not getting the status of a previous request). NOTE: on the second POST we are issuing a user task id (from the first one) but it doesn't exist on the CC side anymore (so it sounds to be kind of useless).

So, I think we should investigate and understand what happens if we use the same rebalancing approach (POST on `/rebalance` and then GET on `/user_tasks`) even in the case of "not enough data in the window" exception. Even because a proposal could be not ready not just because of the "not enough data in window" exception but even for other reasons which don't throw and exception (in that case GETting the user tasks status would be the right way to go).

### PR DESCRIPTION
**Pull request by tinaselenge**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

- Use `/kafkacruisecontrol/user_tasks` endpoint to check the status of proposal request (rebalance request in dry mode) instead of `/kafkacruisecontrol/rebalance` endpoint with userTaskId. This makes it more consistent with how we check the status of rebalance request. 

- Updated MockCruiseControl.setupCCRebalanceResponse() to set user task id to the one that matches verbose response, so that the subsequent requests to /user_tasks endpoint can return a verbose response with load optimisation. 

Closes #10294 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
LGTM assuming the tests pass. Should be reviewed by SMEs (@ppatierno )

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:746]: Genuine question ... I was wondering which case you are trying to address here where there is no sessionId stored in the `KafkaRebalance` custom resource but we are in the `PendingProposal` state.

**Review by ppatierno — COMMENTED** [review]
@tinaselenge I left one comment and overall it seems to be ok. Anyway before merging I would love if you could rebase against the main (where now the auto-rebalancing is available). I would like to give this PR a go locally together with auto-rebalancing before approving.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java
Comment (tinaselenge) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java:746]: We request a new proposal when on `onNew` state. If the response to this request is isNotEnoughDataForProposal, we return PendingProposal state without a sessionID in the rebalance status.

Also the logging on line 741 suggests that we are either requesting a new proposal or checking the status of a submitted proposal. Previously, requestRebalance() would request a new proposal if the sessionId happens to be null. So basically we would be keeping the same behaviour when using getUserTaskStatus().

**Comment by tinaselenge** [comment]
> @tinaselenge I left one comment and overall it seems to be ok. Anyway before merging I would love if you could rebase against the main (where now the auto-rebalancing is available). I would like to give this PR a go locally together with auto-rebalancing before approving.

@ppatierno I have rebased it to main :)

**Comment by ppatierno** [comment]
@tinaselenge thanks! Anyway I would like to wait for this PR https://github.com/strimzi/strimzi-kafka-operator/pull/10652 to be merged and doing another rebase, because it fixes a subtle bug in auto-rebalancing. So I would like to try your PR against the "final" auto-rebalancing. But in general your PR LGTM, just looking for the confirmation by testing it with the auto-rebalancing.

**Comment by ppatierno** [comment]
@tinaselenge finally https://github.com/strimzi/strimzi-kafka-operator/pull/10652 was merged. Can you revise your PR so I can try it locally? Thanks!

**Comment by tinaselenge** [comment]
@ppatierno I have rebased again and addressed the review comment. Thanks

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!
Also tested within the auto-rebalancing scenario and it seems to work.

**Pull request merged**
